### ISSUE TITLE
php dom and xml extension warning, but packages are installed?

### ISSUE THREAD

**Issue by m4ntic0r** [piwik/piwik#12211]
Today i upgraded my server from ubuntu server 17.04 to 17.10 and so far everything seems to be ok but i get system warnings in piwik system check.

There is a warning for the "dom" extension and "simplexml" extension.

On system -> apt-cache search php-dom, the package php7.1-xml is listed and this is installed
On system -> apt-cache search simplexml, the same package php7.1-xml is found and installed

php-xml is installed, too

php version shown in the webui: 7.0.22-0ubuntu0.17.04.1

**Comment by fdellwing** [comment]
You are using `PHP7.0` and not `PHP7.1`.

Please install `php7.0-xml` and `php7.0-dom`.

To check for installed PHP modules run `dpkg -l | grep php`

**Comment by sgiehl** [comment]
Could you create a simple php script to check what `get_loaded_extensions()` returns?

**Comment by fdellwing** [comment]
You mean `php -m` or in this case `php7.0 -m`? ^^

**Comment by m4ntic0r** [comment]
after the error i restored from backup and everything was ok.

now i just tried the upgrade to 17.10 again and now everything seems to run, after upgrade i had to manually install php7.1 and some 7.1 extensions.

system check says everything is ok, and PHP-Version: 7.1.8-1ubuntu1

**Comment by sgiehl** [comment]
> You mean php -m or in this case php7.0 -m? ^^

No, the output of the php function `get_loaded_extensions()`, not sure if that's always the same as php -m

**Comment by fdellwing** [comment]
It is.

**Comment by robocoder** [comment]
`php -m` loads the cli configuration; technically, it could have different extensions loaded than mod_php or php-fpm.

**Comment by BloodyIron** [comment]
Yeah I have php7.2-xml and php7.2-gd installed, yet the checker says they aren't. I'm using the latest version as of today, unsure what on earth it's complaining about.

**Comment by robocoder** [comment]
On Ubuntu, there are separate conf.d/ .ini files for php-cli and mod-apache that load PHP extensions.

You can check this by opening your browser to a file on your system that contains:
```
<?php phpinfo();
```

**Comment by BloodyIron** [comment]
But I know that the modules are loaded as apache shows them enabled (in the modules-enabled folder). Same way all the other modules are loaded. I don't see why the behaviour would be different here.

**Comment by tsteur** [comment]
@BloodyIron  could you test what @robocoder suggests in https://github.com/matomo-org/matomo/issues/12211#issuecomment-573321106 ? It's important to have it tested through the browser. Maybe you need to make some changes.
